lib/connect.js

```diff
@@ -410,6 +414,8 @@ async function connectThroughNode (c, address, socket) {
     serverAddress,
     payload
   }
+  c.encryptedSocket.userData ??= {}
+  c.encryptedSocket.userData.timestamp = payload.timestamp
```

Comment:
is using `userData` ugly? idk. The comment for it in secret-stream says "// pointer for upstream to set data here if they want". Are we upstream? Should we modify secret-stream so it has a timestamp property proper?